src/main/services/job-timeline.ts

```diff
@@ -63,8 +64,15 @@ export class ApiJobTimelineService extends JobTimelineService {
     ) {
         super();
         const baseUrl = this.env.API_JOB_TIMELINE_URL;
-        // subject to change the auth afterwards?
         const authKey = this.env.API_JOB_TIMELINE_KEY;
+        if (!baseUrl || !authKey) {
+            throw new Exception({
+                name: 'ConfigurationError',
+                message: 'Check Environment: API_JOB_TIMELINE_URL, API_JOB_TIMELINE_KEY'
+            });
+        }
+
+        // subject to change the auth afterwards?
```

Comment:
100%, my thoughts as well, we probably need to just pass `Authorization` header to it.

Reply:
This is the case where I struggle with inversify, (and it happens fairly often with this Request!!) 
should we add a method `setAuthorization` and expect a user to always call it before using it to set header? or similar to that, make `request` public so that they can just set the `auth`?